.gitignore

```diff
@@ -8,9 +8,10 @@ target
 Cargo.lock
 
 # Godot (extension list needed to run tests without prior editor opening)
-.godot
 godot.log
-!/itest/.godot/extension_list.cfg
+/itest/godot/.godot/**
+!/itest/godot/.godot/extension_list.cfg
+!/itest/godot/.godot/global_script_class_cache.cfg

```

Comment:
A general `.godot` should stay, we also have other project directories inside `examples`.

Reply:
Once a dir is ignored, you can't un-ignore specific files inside it. You'd want `.godot/**` but then it only matches relative to the current dir.

Solved it in a `.gitignore` per dir since dodge-the-creeps already had one, and it's more maintainable that way (see how the old `!` line was actually wrong).

Reply:
> Once a dir is ignored, you can't un-ignore specific files inside it.

You're right, I just checked [the documentation](https://git-scm.com/docs/gitignore#_pattern_format) and indeed:

> An optional prefix "!" which negates the pattern; any matching file excluded by a previous pattern will become included again. **It is not possible to re-include a file if a parent directory of that file is excluded.**

---

> Solved it in a `.gitignore` per dir since dodge-the-creeps already had one, and it's more maintainable that way

I have mixed feelings about this:
* I have to keep N duplicated `.gitignore` files instead of 1 in sync, where N is the number of Godot projects in the repo (itest + examples). This sounds _less_ maintainable to me.
* Example directories contain one more file which is not part of the example, just there for our own repo organization.

A not-so-nice option is to version files _despite_ being ignored.

I guess `**/.godot/**` as a pattern does not work either?
Or `examples/*/godot/.godot`?

Reply:
Yeah, that works! :+1: